app/src/main/java/com/onix/internship/ui/game/TicTacToeBoard.kt

```diff
@@ -0,0 +1,213 @@
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.content.res.TypedArray
+import android.graphics.Canvas
+import android.graphics.Paint
+import android.util.AttributeSet
+import android.view.MotionEvent
+import android.view.View
+import com.onix.internship.R
+import kotlin.math.ceil
+
+class TicTacToeBoard(context: Context, attrs: AttributeSet?) : View(context, attrs) {
+
+    private val size = 0.2f
+    private var winningLine = false
+    private val paint = Paint()
+    private var cellSize = width / 3
+    private val game = GameViewModel()
+
+
+    private val typedArray: TypedArray = context.theme.obtainStyledAttributes(
+        attrs, R.styleable.TicTacToeBoard, 0, 0
+    )
+    private val boardColor = typedArray.getColor(R.styleable.TicTacToeBoard_boardColor, 0)
+    private val xColor = typedArray.getColor(R.styleable.TicTacToeBoard_XColor, 0)
+    private val oColor = typedArray.getColor(R.styleable.TicTacToeBoard_OColor, 0)
+    private val winningLineColor =
+        typedArray.getColor(R.styleable.TicTacToeBoard_WinningLineColor, 0)
+
+    fun setPlayer(player: Boolean) {
+        if (player) {
+            game.computer()
+        }
+    }
+
+    override fun onMeasure(width: Int, height: Int) {
+        super.onMeasure(width, height)
+
+        val dimension = measuredWidth.coerceAtMost(measuredHeight)
+        cellSize = dimension / 3
+
+        setMeasuredDimension(dimension, dimension)
+    }
+
+    override fun onDraw(canvas: Canvas) {
+        paint.style = Paint.Style.STROKE
+        paint.isAntiAlias = true
+
+        drawGameBoard(canvas)
+        drawMarkers(canvas)
+
+        if (winningLine) {
+            paint.color = winningLineColor
+            drawWinningLine(canvas)
+        }
+    }
+
+    @SuppressLint("ClickableViewAccessibility")
+    override fun onTouchEvent(event: MotionEvent): Boolean {
+        val x = event.x
+        val y = event.y
+
+        val action = event.action
+
+        if (action == MotionEvent.ACTION_DOWN) {
+            val row = ceil(y / cellSize).toInt()
+            val col = ceil(x / cellSize).toInt()
+
+            if (!winningLine) {
+
+                if (game.updateGameBoard(row, col)) {
+                    invalidate()
+                    game.computer()
+                    invalidate()
+                    if (game.winnerCheck()) {
+                        winningLine = true
+                        invalidate()
+                    }
+//                    if (game.player % 2 == 0) {
+//                        game.player = game.player - 1
+//                    } else {
+//                        game.player = game.player + 1
+//                    }
+                }
+            }
+
+            invalidate()
+            return true
+        }
+        return false
+    }
+
+    private fun drawGameBoard(canvas: Canvas) {
+        paint.color = boardColor
+        paint.strokeWidth = 16f
+
+        for (c in 1..2) {
```

Comment:
why 1..2 hardcoded ? digging count of cells from GameViewModel

Reply:
done